### DISCUSSION TITLE
zitadel with google sso

### DISCUSSION THREAD

**Discussion by kioskOG** [zitadel/zitadel#9081] [General]
I'm using self-hosted  zitadel with self-hosted Netbird. I have configured zitadel with google SSO, where i have whitelisted my domain, but i don't want everyone from the domain to be able to access the zitadel, only limited peoples should be allowed. Is this possible ?

**Comment by hifabienne** [comment]
you could add a Zitadel action on the external authentication, where you check for a pre condition. 
A zitadel action is a custom scrip you can run at certain triggers of a process
https://zitadel.com/docs/apis/actions/introduction
